components/b2b/AccountHierarchy/AccountHierarchyEditForm/AccountHierarchyEditForm.tsx

```diff
@@ -0,0 +1,138 @@
+import { useEffect, useState } from 'react'
+
+import { yupResolver } from '@hookform/resolvers/yup'
+import { LoadingButton } from '@mui/lab'
+import { Box, Button, FormControl, InputLabel, MenuItem, Stack, Typography } from '@mui/material'
+import { useTranslation } from 'next-i18next'
+import { useForm, Controller } from 'react-hook-form'
+import * as yup from 'yup'
+
+import { AccountHierarchyFormStyles } from '../AccountHierarchyAddForm/AccountHierarchyAddForm.styles'
+import { KiboSelect } from '@/components/common'
+import { UpdateCustomerB2bAccountParams } from '@/lib/types'
+
+import { B2BAccount } from '@/lib/gql/types'
+
+interface AccountHierarchyEditFormProps {
+  accounts: B2BAccount[]
+  accountToEdit: B2BAccount
+  onSave: (data: UpdateCustomerB2bAccountParams) => void
+  onClose: () => void
+}
```

Comment:
instead of sending accountToEdit just send currentParent i.e. accountToEdit.companyOrOrganization

Reply:
Changed accountToEdit to "b2BAccount", which is more meaningful.
We cannot use currentParent as currentParent is not required to edit parent.